draft-ietf-dnsop-dns-capture-format-01.md

```diff
@@ -232,33 +232,46 @@ It is anticipated
 
 Rationale: Many name servers will process queries on a best-effort basis in accordance with Postel's Law, and do not insist on
-completely well-formed packets. Name servers will also generally ignore any trailing data following well-formed DNS content. Users may wish
-to be informed of such transactions, or input data that cannot be decoded to even a DNS header and which therefore cannot be meaningfully
-processed as part of the Q/R data item stream, and may wish to be able to analyse these malformed inputs as, for example, possible attack
-vectors. Therefore these interactions with the name server should be recorded where possible, but flagged as malformed.
+completely well-formed packets. If possible, responses to these queries should be matched with the query, so that the query does
+not appear to have gone un-answered in name server performance reporting. It can be advantageous to collect statistics on malformed
+inputs, and possibly to analyse those inputs. Therefore it should be possible to record malformed packet contents directly in the C-DNS format.
+
+# Malformed Packets
 
-QUESTION: Should a valid DNS header include additional conditions?
+As the primary aim of handling malformed packets is to match responses and queries as far as possible despite the malformation,
+malformed packets are divided into two distinct types; those packets which can be decoded sufficiently
+to allow matching with a corresponding query or response ("partially malformed"), and those that cannot ("completely malformed").
 
-* The flags and codes words has valid values for operation code and response code.
-* A query has response code 0.
-* All zero bits are set to 0.
+The approach adopted in C-DNS to matching query and response is described in (#matching-algorithm). To be matched, it must be possible
+to decode the Primary ID (see (#primary-id)), and, if present, the Secondary ID (see (#secondary-id)). Briefly, it must be possible
+to decode the IP source and destination addresses, the UDP or TCP source and destination ports, the DNS header ID, the DNS header
+flags/opcode/rcode fields, QDCOUNT, and, if QDCOUNT is non-zero, the entire DNS header and the first question section entry.
+If these cannot be decoded, matching cannot proceed, and the packet is therefore classified as completely malformed. If these can
+be decoded, the packet is classified as partially malformed, and participates in matching. See (#malformed-packet-considerations) for
+more on matching with partially malformed packets.
+
+If recording the content of malformed packets in C-DNS is enabled, the original packet content of both completely and partially
+malformed packets is recorded in C-DNS. The data is tagged to indicate whether the packet was processed as completely or
+partially malformed.
+
+A count of malformed packets is in a block is kept in the C-DNS block statistics. This is a single count, including both
+partially and completely malformed packets.
```

Comment:
SInce you'll always be decoding packets, you'll always know how many malformed packets there are. Both statistics are independent of whether or not you're recording malformed packets. Otherwise, yes, you're right, that a partially malformed packet that matches will be counted both as malformed and a match. I did wonder whether we should have separate counters for partial and complete malformed packets. Or just count completely malformed packets.

What do you think?